lib/record_store/provider/dnsimple.rb

```diff
@@ -14,19 +14,23 @@ def supports_alias?
 
       # returns an array of Record objects that match the records which exist in the provider
       def retrieve_current_records(zone:, stdout: $stdout)
-        session.zones.all_records(account_id, zone).data.map do |record|
-          begin
-            build_from_api(record, zone)
-          rescue StandardError
-            stdout.puts "Cannot build record: #{record}"
-            raise
-          end
-        end.compact
+        retry_on_connection_errors do
```

Comment:
Why put the retries in the providers instead of in the place where we call the providers ?
Would be a single change that affects all ?

Reply:
- For this PR, I only want to patch operations that are already idempotent by design, deferring the effort of making mutable operations idempotent until it'll provide us value.  Patching things where provider methods are called would make it more difficult to guarantee that errors are originating from idempotent operations.  In a future state where all provider methods are retryable, we could move this retry code further down the call stack...even to the command line class. 
- Unit testing is easier this way as I need to test smaller units.  In fact, my original thought was to patch even closer to Net::HTTP (retrying on both timeouts and connection resets based on the request method, sorta like- or piggybacking on-how Net::HTTP's built in retry mechanism works), but besides some monkey patching in this repo, this code that uses Net::HTTP actually exists in DNSimple and NS1 gems.  I'm thinking about perhaps submitting PRs in both repos.  This could make this patch obsolete in the future as these gems would just be resilient by design.  But this PR would still bring value in the meantime.